specs/models/ArtifactSettings.v1.json

```diff
@@ -0,0 +1,157 @@
+{
+  "title": "ArtifactSettings.v1",
+  "type": "object",
+  "description": "Settings for an artifact",
+  "properties": {
+    "customHook": {
```

Comment:
I think these properties make it clear that an "Artifact", as defined here, is really a "Step". 

Our existing (e.g. LoadHubArtifactsCommand and LoadUserArtifactsCommand) code also refers to flows and entity models as artifacts, but those concepts do not support properties like customHook and targetDatabase/sourceDatabase. 

Of course, if this is just capturing the as-is, that's fine, and this is correct. But I think it's critical that we rework the concept of the "artifact service" into a "step service", because we're really describing steps. 

"Artifact" will then continue to have its current meaning of "Config files created by users", which will consist of flows, entity models, steps, step definitions, and mappings (with mappings being separate from mapping steps because of the existing code that expects things to be that way).